### ISSUE TITLE
7.0.0 chart fails on already existing clusterrole

### ISSUE THREAD

**Issue by gavvvr** [appscode/voyager#1092]
Hi. 
If I reinstall voyager locally with my minikube, it prints either warning or error:
> Error: clusterroles.rbac.authorization.k8s.io "appscode:voyager:edit" already exists

Commands to reproduce on clean minikube:

```
helm init
helm install -n voyager-7 --version 7.0.0 --set provider=minikube appscode/voyager
helm del --purge voyager-7
helm install -n voyager-7 --version 7.0.0 --set provider=minikube appscode/voyager
```

With Helm chart for 6.x this message does not appear
> helm install -n voyager-6 --version 3.2.3 --set provider=minikube appscode/voyager

**Comment by tamalsaha** [comment]
What is your `helm version` ?

**Comment by gavvvr** [comment]
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}

**Comment by tamalsaha** [comment]
can you show `kubectl get clusterrole appscode:voyager:edit -o yaml` ?

**Comment by gavvvr** [comment]
> user@solus ~ $ helm del --purge voyager-7
> release "voyager-7" deleted
> user@solus ~ $ kubectl get clusterrole appscode:voyager:edit -o yaml
> apiVersion: rbac.authorization.k8s.io/v1
> kind: ClusterRole
> metadata:
>   annotations:
>     helm.sh/hook: post-install,post-upgrade
>     helm.sh/hook-delete-policy: before-hook-creation
>   creationTimestamp: 2018-06-01T12:43:35Z
>   labels:
>     rbac.authorization.k8s.io/aggregate-to-admin: "true"
>     rbac.authorization.k8s.io/aggregate-to-edit: "true"
>   name: appscode:voyager:edit
>   resourceVersion: "692"
>   selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/appscode%3Avoyager%3Aedit
>   uid: 660b58e5-6599-11e8-9802-0c1183fb614d
> rules:
> - apiGroups:
>   - voyager.appscode.com
>   resources:
>   - certificates
>   - ingresses
>   verbs:
>   - create
>   - delete
>   - deletecollection
>   - get
>   - list
>   - patch
>   - update
>   - watch
> 

I can also see `appscode:voyager:view` after helm purge

**Comment by tamalsaha** [comment]
> I can also see appscode:voyager:view after helm purge

This is by design. When you `helm purge`, it will only delete the operator but leave the actual ingress objects. This cluster role gives users permission to edit them .

**Comment by gavvvr** [comment]
Hmm.

But when I do the same with a previous 3.2.3 chart, there is neither `appscode:voyager:view` or `appscode:voyager:delete` left after purge

**Comment by tamalsaha** [comment]
This is added recently https://github.com/appscode/voyager/blob/master/chart/voyager/templates/user-roles.yaml , since RBAC is default now.

**Comment by gavvvr** [comment]
Thanks @tamalsaha for help.
As you said the problem was solved in [2.9.0 Helm release](https://github.com/kubernetes/helm/releases/tag/v2.9.0):
> Implement before-hook-creation delete policy 1d4883b (Alexey igrychev)

Upgraded helm and checked again, the problem have gone :)
